### PR TITLE
Add first version of create intro page for insight creation UI

### PR DESCRIPTION
**Pull request by vovakulikov**
Resolves https://github.com/sourcegraph/sourcegraph/issues/20200

This PR is the second of three PRs for insight creation UI 
1. [Creation UI for search insight. ](https://github.com/sourcegraph/sourcegraph/pull/20032) https://github.com/sourcegraph/sourcegraph/pull/20032
2. **This Pull Request**
3. [Creation UI for lang stats insights](https://github.com/sourcegraph/sourcegraph/pull/20548)

This PR is about add main entry point for users who want to create they next code insights.
See Figma layouts for more context about this page. 

image

**Note** this PR is based on [first PR for creation UI](https://github.com/sourcegraph/sourcegraph/pull/20032).

### CODE REVIEW

File: client/web/src/insights/pages/creation-intro/CreationIntroPage.tsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+/** Displays intro page for insights creation UI. */
+export const CreationIntroPage: React.FunctionComponent = () => (
+    <Page className="col-8">
+        <PageTitle title="Create code insights" />
+
+        <div className={styles.createIntroPageSubTitleContainer}>
+            <h2>Create new insight</h2>
+
+            <p className="text-muted">
+                Code insights analyse your code based on any search query.{' '}
```
Comment (eseliger) [inline client/web/src/insights/pages/creation-intro/CreationIntroPage.tsx:23]:
```suggestion
                Code insights analyze your code based on any search query.{' '}
```

File: client/web/src/insights/pages/creation-intro/CreationIntroPage.tsx [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+                </a>
+            </p>
+        </div>
+
+        <div className={styles.createIntroPageInsights}>
+            <section className={styles.createIntroPageInsightCard}>
+                <h3>Based on your search query</h3>
+
+                <p>
+                    Search-based insights let you create any data visualisation about your code based on a custom search
```
Comment (eseliger) [inline client/web/src/insights/pages/creation-intro/CreationIntroPage.tsx:39]:
```suggestion
                    Search-based insights let you create any data visualization about your code based on a custom search
```

**Review by eseliger — APPROVED** [review]

**Comment by vovakulikov** [comment]
I created [a story](https://5f0f381c0e50750022dc6bf7-wnzbtdnann.chromatic.com/?path=/story/web-insights-creationinsightintropage--page) for visual review @AlicjaSuska fyi

**Comment by vovakulikov** [comment]
@Joelkw if you got a moment can you take a look at texts for this page

**Comment by Joelkw** [comment]
@vovakulikov yep, am looking at it now!

**Review by Joelkw — APPROVED** [review]
Copy notes inline! (I know we'll still be tweaking designs + adding extension icons or similar.)

File: client/web/src/insights/pages/creation-intro/charts-mock.ts [hunk 3]
```diff
@@ -0,0 +1,97 @@
+import { LineChartContent, PieChartContent } from 'sourcegraph'
+
+/** Mock data for search-based insight preview for create-intro page. */
+export const LINE_CHART_DATA: LineChartContent<any, string> = {
+    chart: 'line',
+    data: [
+        {
+            date: 1597698000000,
+            'Imports of old GQL.* types': 250,
```
[the remark this answers is not in the document]
Comment (vovakulikov) [reply client/web/src/insights/pages/creation-intro/charts-mock.ts:9]: @Joelkw I believe that's right paring here (at some point we have one tool for `graphQL` generated types and now we have another thing that stores all types in `graphql-operations` files)

But the example itself probably too specific I mean I took this example from your search insight from sourcegraph.com. Maybe we need to come up with some generic examples here.

[the remark this answers is not in the document]
Comment (Joelkw) [reply client/web/src/insights/pages/creation-intro/charts-mock.ts:9]: Hmm, I defer to you and @felixfbecker here! I think more important than being generic is that it is "recognizable" (a `GQL.*` that gets replaced by something like `graphql-operations` is different in "format" so it's less obvious to a user that we're tracking replacements – something like `oldClass.*` being replaced by `newClass.*` might make more sense here)?

**Pull request merged**